Controller/Webhook/Index.php

```diff
@@ -11,352 +11,135 @@
-     * @param HmacSignature $hmacSignature
      * @param NotificationReceiver $notificationReceiver
-     * @param RemoteAddress $remoteAddress
-     * @param AdyenNotificationRepositoryInterface $notificationRepository
+     * @param WebhookAcceptorFactory $webhookAcceptorFactory
      */
     public function __construct(
         Context $context,
-        private readonly NotificationFactory $notificationFactory,
-        private readonly Data $adyenHelper,
         private readonly AdyenLogger $adyenLogger,
-        private readonly SerializerInterface $serializer,
         private readonly Config $configHelper,
-        private readonly IpAddress $ipAddressHelper,
-        private readonly RateLimiter $rateLimiterHelper,
-        private readonly HmacSignature $hmacSignature,
         private readonly NotificationReceiver $notificationReceiver,
-        private readonly RemoteAddress $remoteAddress,
-        private readonly AdyenNotificationRepositoryInterface $notificationRepository
+        private readonly WebhookAcceptorFactory $webhookAcceptorFactory
     ) {
         parent::__construct($context);
-
-        // Fix for Magento2.3 adding isAjax to the request params
-        if (interface_exists(CsrfAwareActionInterface::class)) {
-            $request = $this->getRequest();
-            if ($request instanceof Http && $request->isPost()) {
-                $request->setParam('isAjax', true);
-                $request->getHeaders()->addHeaderLine('X_REQUESTED_WITH', 'XMLHttpRequest');
-            }
-        }
+        $this->enforceAjaxHeaderForMagento23Compatibility();
     }
 
     /**
      * @throws LocalizedException
      */
     public function execute(): void
     {
-        // Read JSON encoded notification body
         $notificationItems = json_decode((string) $this->getRequest()->getContent(), true);
 
-        // Check notification mode
-        if (!isset($notificationItems['live'])) {
+        if (!$this->isNotificationModeValid($notificationItems)) {
             $this->return401();
             return;
         }
-        $notificationMode = $notificationItems['live'];
-        $demoMode = $this->configHelper->isDemoMode();
-        if (!$this->notificationReceiver->validateNotificationMode($notificationMode, $demoMode)) {
-            throw new LocalizedException(
-                __('Mismatch between Live/Test modes of Magento store and the Adyen platform')
-            );
-        }
 
         try {
-            // Process each notification item
             $acceptedMessage = '';
-            foreach ($notificationItems['notificationItems'] as $notificationItem) {
-                $status = $this->processNotification(
-                    $notificationItem['NotificationRequestItem'],
-                    $notificationMode
-                );
 
-                if ($status !== true) {
+            foreach ($notificationItems['notificationItems'] as $notificationItemWrapper) {
```

Comment:
Token lifecycle webhooks do not have `notificationItems` array field. This block might fail for non-standard webhook.